### PR TITLE
fix issue #5225

### PR DESCRIPTION
**Pull request by rahu**
when doing a offline log:
- update Geocache members
- refresh View

### CODE REVIEW

**Comment by Lineflyer** [comment]
Does someone have the chance to review this PR because a fix to this issue would be highly appreciated before being able to release from `master`

File: main/src/cgeo/geocaching/ui/LoggingUI.java [hunk 1]
```diff
@@ -114,6 +118,13 @@ public void onClick(final DialogInterface dialog, final int item) {
                 } else {
                     cache.logOffline(activity, logTypeEntry.logType);
                 }
+                if (selectedView != null) {
+                    final ViewHolder holder;
+                    holder = (ViewHolder) selectedView.getTag();
+					if (holder != null) {
```
Comment (samueltardieu) [inline main/src/cgeo/geocaching/ui/LoggingUI.java]: Identation problem.

File: main/src/cgeo/geocaching/ui/LoggingUI.java [hunk 2]
```diff
@@ -114,6 +118,13 @@ public void onClick(final DialogInterface dialog, final int item) {
                 } else {
                     cache.logOffline(activity, logTypeEntry.logType);
                 }
+                if (selectedView != null) {
+                    final ViewHolder holder;
+                    holder = (ViewHolder) selectedView.getTag();
```
Comment (samueltardieu) [inline main/src/cgeo/geocaching/ui/LoggingUI.java]: Please initialize while declaring.

File: main/src/cgeo/geocaching/ui/CacheListAdapter.java [hunk 3]
```diff
@@ -426,8 +430,8 @@ public View getView(final int position, final View rowView, final ViewGroup pare
         else {
             holder.text.setText(cache.getName());
         }
-        holder.text.setCompoundDrawablesWithIntrinsicBounds(MapUtils.getCacheMarker(res, cache, cacheListType), null, null, null);
-
+        holder.cacheListType = cacheListType;
+        updateViewHolder(holder, cache, res);
```
Comment (samueltardieu) [inline main/src/cgeo/geocaching/ui/CacheListAdapter.java]: Wouldn't it be enough to pass `cacheListType` to `updateViewHolder`? Does it need to be stored in the holder itself?

Comment (rahu) [inline main/src/cgeo/geocaching/ui/CacheListAdapter.java]: cacheListType is needed in updateViewHolder. This is now called from  main/src/cgeo/geocaching/ui/LoggingUI.java where I can get the holder but not the CacheListType. Therefore cacheListType needs to be stored in the holder.

**Comment by samueltardieu** [comment]
Please squash your commits together.

File: main/src/cgeo/geocaching/ui/CacheListAdapter.java (hunk 3 above)
Comment (samueltardieu) [inline main/src/cgeo/geocaching/ui/CacheListAdapter.java]: Ok. I couldn't see that easily by just reading the diff.

**Comment by samueltardieu** [comment]
Squashed and merged.

**Comment by samueltardieu** [comment]
Note: this should have been based on the `release` branch as to be mergeable with the next bugfix release. Please look at https://github.com/cgeo/cgeo/wiki/How-to-get-a-bug-fix-into-the-release.

Thanks for your contribution.

**Pull request closed without merging**